src/map/script.c

```diff
@@ -20365,6 +20387,45 @@ static BUILDIN(setunitname)
 	return true;
 }
 
+static BUILDIN(setunittitle)
+{
+	struct block_list *bl = map->id2bl(script_getnum(st, 2));
+	if (bl == NULL) {
+		ShowWarning("buildin_setunittitle: Error in finding object with given game ID %d!\n", script_getnum(st, 2));
+		return false;
+	}
+
+	struct unit_data *ud = unit->bl2ud2(bl);
+	if (ud == NULL) {
+		ShowWarning("buildin_setunittitle: Error in finding unit_data for given game ID %d!\n", script_getnum(st, 2));
+		return false;
+	}
+
```

Comment:
the documentation said it doesn't work for players
then it should throw error on `if (bl->type == BL_PC)`

Reply:
technically it can works for players too. But for now useless.
As invisible option can works even now. As visible need add support.

Reply:
it works on npc, but needs to move out-of-sight and move-in-sight again to see it
probably missing `clif->blname_ack(0, bl);`

```c
prontera,155,185,5	script	kjhsdff	1_F_MARIA,{
//	setunittitle getcharid(3), "asdf"; // makes my name disappear
	setunittitle getnpcid(), "asdf";
//	.@mobgid = monster("this", -1,-1, "--ja--", PORING,1 );
//	setunittitle .@mobgid, "asdf";
}
```
why it makes my name disappear ?
also does it works on monster/pets ? just test it doesn't really do anything

Reply:
yes, here for now no name update. i will add it
for mobs because it using additional fields with hp, for now mobs using other packet

Reply:
for pets, homun, mercenary it should works now